File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        if(headA == nullptr || headB == nullptr){
            return nullptr;
        }

        ListNode* crrA = headA;
        while(crrA != nullptr){
             ListNode* crrB = headB;
            while(crrB != nullptr){
                if(crrA == crrB){
                    return crrA;
                }
                crrB = crrB->next;
            }
            crrA = crrA->next;
        }
    return nullptr;
    }
};

File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        if (headA == null || headB == null) {
            return null;
        }

        ListNode crrFirst = headA;
        ListNode crrSecond = headB;

        while (crrFirst != crrSecond) {
            crrFirst = (crrFirst == null) ? headB : crrFirst.next;
            crrSecond = (crrSecond == null) ? headA : crrSecond.next;
        }

        return crrFirst;
    }
}